File: Codeforces/gym/101845/m.py
#coding: utf-8


# cunfused M and N order
M, N, K = map(int, raw_input().split())

prob = 0.0

if (K <= N):   # fill K-1 with blue
    N -= K-1
    prob = (K-1)
    prob += float(N) / float(N+M)
else:
    prob = float(N)
    N = 0
    M -= (K - N) - 1

prob /= float(K)

print "%.15f" % prob

File: Codeforces/gym/101845/m.cpp
#include <bits/stdc++.h>

using namespace std;

int main() {
    int M, N, K;    //confused N and M, so changed order
    scanf(" %d %d %d", &M, &N, &K);

    double prob = 0.0;

    if (K <= N){   // fill K-1 with blue
        N -= K-1;
        prob = (K-1);
        prob += ((double)(N)) / ((double) (N+M));
    } else {       // spend all blue
        prob = (double) (N);
    }

    prob /= (double) (K);

    printf("%.15lf\n", prob);
    return 0;
}
